### DISCUSSION TITLE
Regular Expression Matching in Workflow Conditions

### DISCUSSION THREAD

**Discussion by mio4kon** [argoproj/argo-workflows#13677] [Q&A] [answered]
I am currently working with Argo Workflows and trying to utilize regular expressions in the when conditions of my workflow tasks. Specifically, I am attempting to match a branch name that starts with "release" using the following expression:

`when: "{{=jsonpath(workflow.parameters.metaData, '$.repoInfo.branch') =~ '^release.*'}}"`

However, I am encountering errors related to the syntax, such as `Invalid token: '{{='.` It seems that the use of `=` before the expression is causing issues.

I would like to clarify the correct way to use regular expressions in the when conditions. Is there a specific syntax or method that I should follow to achieve this functionality?

![image](https://github.com/user-attachments/assets/569b00ff-6ed8-46bc-bfbe-9ac423cf1938)

![image](https://github.com/user-attachments/assets/6d602eea-8cfd-4191-a949-3deb29bae156)

**Comment by jswxstw** [comment, marked as answer]
The operator of regex in [expr-lang](https://expr-lang.org/docs/language-definition) is `matches`, please try as below:

```YAML
when: "{{=jsonpath(workflow.parameters.metaData, '$.repoInfo.branch') matches '^release.*'}}"
```

**Reply by mio4kon** [reply]
Thanks！
